src/style-spec/reference/v8.json

```diff
@@ -2606,7 +2606,7 @@
         }
       },
       "feature-state": {
-        "doc": "Retrieves a property value from the current feature's state. Returns null if the requested property is not present on the feature's state object.",
+        "doc": "Retrieves a property value from the current feature's state. Returns null if the requested property is not present on the feature's state. A feature's state is not part of the GeoJSON or vector tile data, and must be set programmatically on each feature. Note that [\"feature-state\"] can only be used with paint properties that support data-driven styling.",
```

Comment:
> A feature's state is not part of the GeoJSON or vector tile data, and must be set programmatically on each feature.

@anandthakker @samanpwbb Does this statement work for capturing that feature state must be set programmatically on each session? Not sure if the docs should also mention that the state doesn't persist across sessions/reloads of the map.